python/lsst/analysis/tools/atools/stellarLocus.py

```diff
@@ -54,239 +55,259 @@ def setDefaults(self):
 
-        self.prep.selectors.starSelector = StarSelector()
+        self.prep.selectors.magSelector = MagSelector()
+        self.prep.selectors.magSelector.maxMag = 22.5
+
+        self.prep.selectors.starSelector1 = StarSelector()
+        self.prep.selectors.starSelector2 = StarSelector()
+        self.prep.selectors.starSelector3 = StarSelector()
 
         self.process.buildActions.x = ExtinctionCorrectedMagDiff()
         self.process.buildActions.x.magDiff.returnMillimags = False
         self.process.buildActions.y = ExtinctionCorrectedMagDiff()
         self.process.buildActions.y.magDiff.returnMillimags = False
 
+        self.process.calculateActions.approxMagDepth = ApproxFloor(vectorKey="mag")
 
-class WPerpPSF(StellarLocusBase):
-    parameterizedBand: bool = False
+        self.produce.plot = ColorColorFitPlot()
 
+
+class WPerpPSF(StellarLocusBase):
     def setDefaults(self):
         super().setDefaults()
-        self.prep.selectors.flagSelector.bands = ["g", "r", "i"]
+        colorBands = ["g", "r", "i"]
+        fluxType = "psfFlux"
+        self.prep.selectors.flagSelector.bands = colorBands
         self.prep.selectors.snSelector.bands = ["r"]
-        self.prep.selectors.snSelector.fluxType = "{band}_psfFlux"
-
-        self.prep.selectors.starSelector.vectorKey = "r_extendedness"
+        self.prep.selectors.snSelector.fluxType = "{band}_" + fluxType
+        self.prep.selectors.magSelector.bands = ["r"]
+        self.prep.selectors.magSelector.fluxType = "{band}_" + fluxType
 
-        self.process.buildActions.x.magDiff.col1 = "g_psfFlux"
-        self.process.buildActions.x.magDiff.col2 = "r_psfFlux"
+        self.prep.selectors.starSelector1.vectorKey = colorBands[0] + "_extendedness"
+        self.prep.selectors.starSelector2.vectorKey = colorBands[1] + "_extendedness"
+        self.prep.selectors.starSelector3.vectorKey = colorBands[2] + "_extendedness"
 
-        self.process.buildActions.y.magDiff.col1 = "r_psfFlux"
-        self.process.buildActions.y.magDiff.col2 = "i_psfFlux"
-
-        self.process.buildActions.mag = ConvertFluxToMag(vectorKey="r_psfFlux")
+        self.process.buildActions.x.magDiff.col1 = colorBands[0] + "_" + fluxType
+        self.process.buildActions.x.magDiff.col2 = colorBands[1] + "_" + fluxType
+        self.process.buildActions.y.magDiff.col1 = colorBands[1] + "_" + fluxType
+        self.process.buildActions.y.magDiff.col2 = colorBands[2] + "_" + fluxType
+        self.process.buildActions.mag = ConvertFluxToMag(vectorKey=colorBands[1] + "_" + fluxType)
 
         self.process.calculateActions.wPerp_psfFlux = StellarLocusFitAction()
-        self.process.calculateActions.wPerp_psfFlux.stellarLocusFitDict = {
+        stellarLocusFitDict = {
             "xMin": 0.28,
             "xMax": 1.0,
             "yMin": 0.02,
             "yMax": 0.48,
-            "mHW": 0.52,
-            "bHW": -0.08,
+            "mFixed": 0.483,
```

Comment:
Where did these numbers come from?  You should make a comment in the code with a link to any relevant analysis.

Reply:
Yes please, it's much more obvious there.